style.json

```diff
@@ -394,13 +351,12 @@
       },
       "source": "basemap",
       "source-layer": "landcover",
-      "filter": [
-        "all",
-        ["==", "class", "grass"],
-        ["!in", "subclass", "park", "garden"]
-      ],
+      "filter": ["==", "class", "grass"],
       "paint": {
-        "fill-color": "#e0f2d3"
+        "fill-color": ["case",
+          ["in", ["get", "subclass"], ["literal", ["park", "garden"]]], "#c0eab8",
+          "#e0f2d3"
+        ]
```

Comment:
Note that merging these 2 layers may cause some changes in the rendering if some features are placed on top of each other, depending on their order in the vector tile.
But that problem should be solved on the vector tile generation, for example by reordering the polygons by descending area.